boot/bootstate20.go

```diff
@@ -496,6 +496,9 @@ type bootState20Base struct{}
 // revisions returns the current boot snap and optional try boot snap for the
 // type specified in bsgeneric.
 func (bs20 *bootState20Base) revisions() (curSnap, trySnap snap.PlaceInfo, tryingStatus string, err error) {
+	modeenvLock()
+	defer modeenvUnlock()
+
```

Comment:
we can't fix it this way,  as other places call this with the lock taken afaik. As I said I think the issue is boot.InUse, but we need to add a unit test using that with a core20 setup